### PR TITLE
[SPARK-25296][SQL][TEST] Create ExplainSuite

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
Move the output verification of Explain test cases to a new suite ExplainSuite.

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
LGTM, this is cleaner!

File: sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.apache.spark.sql.test.SharedSQLContext
+import org.apache.spark.sql.types.StructType
+
+class ExplainSuite extends QueryTest with SharedSQLContext {
+  import testImplicits._
+
+  /**
+   * Runs the plan and makes sure the plans contains all of the keywords.
+   */
+  private def checkKeywordsExistsInExplain(df: DataFrame, keywords: String*): Unit = {
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala]: Can't we move this helper func into `QueryTest`?

**Comment by kiszk** [comment]
SGTM. can we use this utility function in `StreamSuite`, too?

File: sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala (hunk 1 above)
Comment (gatorsmile) [reply sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala]: This is EXPLAIN specific verification. In the next release, we need to further enhance EXPLAIN. I think this helper function will be changed a lot.

If this need to be shared with the other test suites, we can move it to `QueryTest`.

**Comment by gatorsmile** [comment]
@kiszk Could you help move the related test cases in StreamSuite.scala to here?

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Comment by maropu** [comment]
@kiszk ping, could you do that? https://github.com/apache/spark/pull/22300#issuecomment-417706754

**Pull request closed without merging**